lib/s3middleware/validateConditionalHeaders.js

```diff
@@ -68,6 +68,31 @@ function _checkUnmodifiedSince(ifUnmodifiedSinceTime, lastModified) {
     return res;
 }
 
+/**
+ * checks 'if-modified-since' and 'if-unmodified-since' headers if included in
+ * request against last-modified date of object
+ * @param {object} headers - headers from request object
+ * @param {string} lastModified - last modified date of object
+ * @return {object} contains modifiedSince and unmodifiedSince res objects
+ */
+function checkDateModifiedHeaders(headers, lastModified) {
+    let lastModifiedDate = new Date(lastModified);
+    lastModifiedDate.setMilliseconds(0);
+    lastModifiedDate = lastModifiedDate.getTime();
```

Comment:
You should probably send `lastModifiedDate` as a parameter so that this method can be used as a true utility.

Reply:
I'm not sure what you mean by this... I put the `lastModifiedDate` manipulation here so that it wouldn't have to be repeated in the `validateConditionalHeaders` function and the `objectDelete` api

Reply:
So instead of checking it against current date it would make it easy to check it against any date. Writing different cases for unit tests mimicking old dates would be easier. The intention of checking utilities in Arsenal is that it may be used by many projects (S3, Backbeat, Connect etc.) and since it's hard to anticipate every need making a utility as flexible as possible would help.

Reply:
Discusses, change unneeded